### PR TITLE
Enable code coverage on Travis CI

### PR DESCRIPTION
**Pull request by sergeyklay**
I have no idea how to enable code coverage on OSX. Looks like we'll need `brew install lcov` or something like that:
```
clang  -lgcov --coverage  -o beanstalkd darwin.o conn.o file.o heap.o job.o ms.o net.o primes.o prot.o sd-daemon.o serv.o time.o tube.o util.o vers.o walg.o main.o 
ld: library not found for -lgcov
clang: error: linker command failed with exit code 1 (use -v to see invocation)
make: *** [beanstalkd] Error 1
```

Another weird thing when use Clang on Linux:
```
./vers.gcno:version '402*', prefer '408*'
./vers.gcno:no functions found
./ct.gcno:version '402*', prefer '408*'

gcov: out of memory allocating 14955062000 bytes after a total of 135168 bytes
./_ctcheck.gcno:version '402*', prefer '408*'
./_ctcheck.gcno:no functions found
./file.gcno:version '402*', prefer '408*'

gcov: out of memory allocating 13612862216 bytes after a total of 135168 bytes
./testserv.gcno:version '402*', prefer '408*
...
...
...
```

So I enable it only for Linux and GCC for now. FYI: Current code coverage: 32.71%

Closes: #420, #396

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -12,10 +12,24 @@ os:
...
-    - export TRAVIS_COMMIT_LOG=`git log --format=fuller -2`
+
+install:
+    - |
+    # Currently works only with gcc on linux
+      if [[ "${TRAVIS_OS_NAME}" == "linux" ]]; then
+          if [[ "${TRAVIS_COMPILER} == "gcc" "]]; then
+              export LDFLAGS=" -lgcov --coverage"
+              export CFLAGS="-O0 -ggdb -fprofile-arcs -ftest-coverage"
+              export CXXFLAGS="-O0 -ggdb -fprofile-arcs -ftest-coverage"
```
Comment (ysmolsky) [inline .travis.yml]: Why do we need CXXFLAGS?

File: .travis.yml [hunk 2]
```diff
@@ -26,4 +40,15 @@ after_script:
     - printf "$TRAVIS_COMMIT_LOG\n"
 
 after_success:
-    - if [[ ! -z "${CODECOV_TOKEN}" ]]; then (bash <(curl -s https://codecov.io/bash) || echo "Codecov did not collect coverage reports"); fi;
```
Comment (ysmolsky) [inline .travis.yml]: Why did you remove this check? if CODECOV_TOKEN is removed from travis setting, the script would fail.

File: .travis.yml [hunk 3]
```diff
@@ -26,4 +40,15 @@ after_script:
...
+    - |
+      if [[ "${TRAVIS_OS_NAME}" == "linux" ]]; then
+          if [[ "${TRAVIS_COMPILER} == "gcc" "]]; then
+              curl -sSl https://codecov.io/bash -o "./codecov"
+              chmod +x codecov
+              ./codecov
+          else
+              echi "Skip collecting coverage reports for clang"
```
Comment (ysmolsky) [inline .travis.yml]:
```suggestion
              echo "Skip coverage reports for clang"
```

File: .travis.yml [hunk 4]
```diff
@@ -26,4 +40,15 @@ after_script:
...
+          fi
+      else
+          echo "Codecov did not collect coverage reports"
```
Comment (ysmolsky) [inline .travis.yml]:
```suggestion
          echo "Skip coverage reports for OSX"
```

File: .travis.yml [hunk 5]
```diff
@@ -12,10 +12,24 @@ os:
...
 before_install:
     - $CC --version
-    - export MAKEJOBS="-j$(getconf _NPROCESSORS_ONLN)"
-    - export TRAVIS_COMMIT_LOG=`git log --format=fuller -2`
+
+install:
+    - |
+    # Currently works only with gcc on linux
+      if [[ "${TRAVIS_OS_NAME}" == "linux" ]]; then
+          if [[ "${TRAVIS_COMPILER} == "gcc" "]]; then
```
Comment (ysmolsky) [inline .travis.yml]: Redundant quote?
```suggestion
          if [[ "${TRAVIS_COMPILER} == "gcc" ]]; then
```

File: .travis.yml [hunk 6]
```diff
@@ -26,4 +40,15 @@ after_script:
     - printf "$TRAVIS_COMMIT_LOG\n"
 
 after_success:
-    - if [[ ! -z "${CODECOV_TOKEN}" ]]; then (bash <(curl -s https://codecov.io/bash) || echo "Codecov did not collect coverage reports"); fi;
+    - |
+      if [[ "${TRAVIS_OS_NAME}" == "linux" ]]; then
+          if [[ "${TRAVIS_COMPILER} == "gcc" "]]; then
```
Comment (ysmolsky) [inline .travis.yml]: Redundant quote too?
```suggestion
          if [[ "${TRAVIS_COMPILER} == "gcc" ]]; then
```

**Comment by ysmolsky** [comment]
> So I enable it only for Linux and GCC for now. FYI: Current code coverage: 32.71%

Having it on Linux is good enough for a start.

File: .travis.yml (hunk 2 above)
Comment (sergeyklay) [reply .travis.yml]: As I realized it needed only for private repos. As you can see it works w/o token now.

**Comment by sergeyklay** [comment]
@ysmolsky Done. For some reason, the build does not start.

File: .travis.yml (hunk 2 above)
Comment (ysmolsky) [reply .travis.yml]: Right on. I was confused too.

**Comment by sergeyklay** [comment]
@ysmolsky Done

**Comment by sergeyklay** [comment]
Here is a badge: [codecov](https://codecov.io/gh/beanstalkd/beanstalkd)

```md
[![codecov](https://codecov.io/gh/beanstalkd/beanstalkd/branch/master/graph/badge.svg)](https://codecov.io/gh/beanstalkd/beanstalkd)
```
